docs/components/form-associated.md

```diff
@@ -123,10 +123,183 @@ the `ElementInternals` API, including [setting the element's
 validity](https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals/setValidity),
 reading the validity state of the form, reading other form values, and more.
 
+## Lifecycle Callbacks
+
+Stencil allows developers building form-associated custom elements to define a
+[standard series of lifecycle
+callbacks](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-reactions)
+which enable their components to react dynamically to events in their
+lifecycle. These could allow fetching data when a form loads, changing styles
+when a form's `disabled` state is toggled, resetting form data cleanly, and more.
+
+### `formAssociatedCallback`
+
+This callback is called when the browser both associates the element with and
+disassociates the element from a form element. The function is called with the
+form element as an argument. This could be used to set an `ariaLabel` when the
+form is ready to use, like so:
+
+```tsx title='src/components/form-associated-cb.tsx'
+import { Component, h, AttachInternals } from '@stencil/core';
+
+@Component({
+  tag: 'form-associated',
+  formAssociated: true,
```

Comment:
Non blocking - The other examples here set `shadow: true,` here as well. Should we add it here for consistency? IDK if folks will try to find some meaning in its absence here or not

Reply:
hmmmmmm yeah I think for consistency I'll just remove it on all of them